compiler/testData/codegen/box/diagnostics/functions/tailRecursion/defaultArgsWithSideEffects.kt

```diff
@@ -0,0 +1,34 @@
+// DONT_RUN_GENERATED_CODE: JS
+// IGNORE_BACKEND: JVM
+
+var counter = 0
+fun inc() = counter++
+
+tailrec fun test(x: Int = 0, y: Int = inc(), z: Int = inc()) {
+    if (x * 2 != y || z - y != 1)
+        throw IllegalArgumentException()
```

Comment:
Please add something like `"x=$x y=$y z=$z"` to the exception message here and in other tests